Gordon360/Services/RecIM/SeriesService.cs

```diff
@@ -0,0 +1,583 @@
+            return s;
+        }
+        public async Task UpdateSeriesTeamStats(SeriesTeamPatchViewModel update)
+        {
+            var st = await _context.SeriesTeam.FindAsync(update.ID);
+            st.Win = update.Win ?? st.Win;
+            st.Loss = update.Loss ?? st.Loss;
+
+            await _context.SaveChangesAsync();
+        }
+
+        private async Task CreateSeriesTeamMappingAsync(IEnumerable<int> teams, int seriesID)
+        {
+            foreach (var teamID in teams)
+            {
+                var seriesTeam = new SeriesTeam
+                {
+                    TeamID = teamID,
+                    SeriesID = seriesID,
+                    Win = 0,
+                    Loss = 0
+                };
+                await _context.SeriesTeam.AddAsync(seriesTeam);
+            }
+            await _context.SaveChangesAsync();
+        }
+
+        public async Task DeleteSeriesCascadeAsync(int seriesID)
+        {
+            //delete series teams
+            var seriesTeam = _context.SeriesTeam.Where(st => st.SeriesID == seriesID);
+            _context.SeriesTeam.RemoveRange(seriesTeam);
+            //delete series surfaces
+            var seriesSurface = _context.SeriesSurface.Where(ss => ss.SeriesID == seriesID);
+            _context.SeriesSurface.RemoveRange(seriesSurface);
+            //delete matches
+            var matchIDs = _context.Match.Where(m => m.SeriesID == seriesID).Select(m => m.ID).ToList();
+            foreach (var matchID in matchIDs)
+            {
+                await _matchService.DeleteMatchCascadeAsync(matchID);
+            }
+            //delete series
+            var series = _context.Series.FirstOrDefault(s => s.ID == seriesID);
+            _context.Series.Remove(series);
+            await _context.SaveChangesAsync();
+        }
+
+        // Scheduler does not currently handle overlaps
+        // eventually:
+        // - ensure that matches that occur within 1 hour do not share the same surface
+        //    unless they're in the same series
+        public async Task<IEnumerable<MatchViewModel>?> ScheduleMatchesAsync(int seriesID)
+        {
+            var series = _context.Series
+                    .FirstOrDefault(s => s.ID == seriesID);
+            var typeCode = _context.SeriesType
+                .FirstOrDefault(st =>
+                    st.ID == series.TypeID
+                ).TypeCode;
+
+            if (typeCode == "RR")
+            {
+                return await ScheduleRoundRobin(seriesID);
+            }
+            if (typeCode == "SE")
+            {
+                return await ScheduleSingleElimination(seriesID);
+            }
+            if (typeCode == "DE")
+            {
+                return await ScheduleDoubleElimination(seriesID);
+            }
+            if (typeCode == "L")
+            {
+                return await ScheduleLadderAsync(seriesID);
+            }
+            return null;
```

Comment:
This would be better as a `switch` expression

Reply:
Agreed, I'll implement it

Reply:
It looks like you changed this to a `switch` statement, which is potentially better. But a `switch` expression works even better:
```suggestion
            return typeCode switch {
              "RR" => await ScheduleRoundRobin(seriesID),
              "SE" => await ScheduleSingleElimination(seriesID),
              "DE" => await ScheduleDoubleElimination(seriesID),
              "L" => await ScheduleLadderAsync(seriesID),
              _ => null
            }
```

Reply:
I must have forgotten to push? I quite literally did that exact change last night.

Reply:
I see that you pushed the switch **statement** version (using `case` and having to return from each case), so I'm just recommending that you simplify further to the switch **expression** version.

Reply:
Right, but what I'm saying is that https://github.com/gordon-cs/gordon-360-api/pull/733/commits/bf13e7fdfe481da9501554a372a54cd82ab6c2f1 already did that exact change. I don't know why it didn't show up for you or say that it's outdated

![image](https://user-images.githubusercontent.com/78386128/222907754-77563aad-ff1d-4071-93d7-62a078e00ec8.png)

Reply:
A certified github moment for sure

Reply:
Oh you're right, I must have missed that, sorry. I think the issue is that I was looking at your commits in order and noticed that you changed to the switch statement version and made this comment. But then I didn't notice that you later changed it to a switch expression so I didn't correct this. Sorry for the confusion.